src/timer.c

```diff
@@ -94,15 +93,22 @@ int uv_timer_start(uv_timer_t* handle,
   return 0;
 }
 
-
-int uv_timer_stop(uv_timer_t* handle) {
-  if (!uv__is_active(handle))
-    return 0;
-
+static void timer_stop(uv_timer_t* handle) {
   heap_remove(timer_heap(handle->loop),
               (struct heap_node*) &handle->heap_node,
               timer_less_than);
   uv__handle_stop(handle);
+}
+
+
+int uv_timer_stop(uv_timer_t* handle) {
+  if (handle->flags & UV_HANDLE_TIMER_READY) {
+    handle->flags &= ~UV_HANDLE_TIMER_READY;
+    uv__queue_remove(&handle->queue_node);
```

Comment:
```suggestion
    uv__queue_remove(&handle->queue_node);
    assert(!uv__is_active(handle));
    return 0;
```
It _looks_ like it can fall through and clobber `handle->heap_node` but IIUC the handle is always stopped when the flag is set, right?

Reply:
Yeah.